Source/TrackableEntities.EF.5/DbContextExtensions.cs

```diff
@@ -199,16 +199,31 @@ public static void ApplyChanges(this InterceptorPool pool, IEnumerable<ITrackabl
                     context.ApplyChangesOnProperties(item, visitationHelper.Clone(), TrackingState.Deleted, interceptors);
                 }
 
+                var isComplex = context.IsComplexType(item.GetType());
+                bool isParentEligible(TrackingState parentState) => parentState == TrackingState.Modified || parentState == TrackingState.Unchanged;
+                var isModifiable = 
+                          (isComplex 
+                            ? isParentEligible(parent.TrackingState) 
+                            : (item.TrackingState == TrackingState.Modified && (state == null || state == TrackingState.Modified)))                          
+                          && item.ModifiedProperties?.Count > 0;
+
                 // Set modified properties
-                if (item.TrackingState == TrackingState.Modified
-                    && (state == null || state == TrackingState.Modified)
-                    && item.ModifiedProperties != null
-                    && item.ModifiedProperties.Count > 0)
+                if (isModifiable)
```

Comment:
I actually created a method already, then I thought I wouldn't want to pollute your class with more method so I stuck it all inline, I'm glad you're also voting for extraction. Done.